src/core/lib/security/transport/client_auth_filter.cc

```diff
@@ -89,16 +91,16 @@ struct call_data {
   void destroy() {
     grpc_credentials_mdelem_array_destroy(&md_array);
     creds.reset();
-    grpc_slice_unref_internal(host);
-    grpc_slice_unref_internal(method);
     grpc_auth_metadata_context_reset(&auth_md_context);
+    host.Destroy();
+    method.Destroy();
   }
 
   grpc_call_stack* owning_call;
   grpc_core::CallCombiner* call_combiner;
   grpc_core::RefCountedPtr<grpc_call_credentials> creds;
-  grpc_slice host = grpc_empty_slice();
-  grpc_slice method = grpc_empty_slice();
+  grpc_core::ManualConstructor<grpc_core::Slice> host;
```

Comment:
The following comment made me believe otherwise (on line 84 in the original):
```
  // This method is technically the dtor of this class. However, since
  // `get_request_metadata_cancel_closure` can run in parallel to
  // `destroy_call_elem`, we cannot call the dtor in them. Otherwise,
  // fields will be accessed after calling dtor, and msan correctly complains
  // that the memory is not initialized.
  void destroy() {
```

Reply:
Interesting.  I suspect there's probably a bug somewhere here that's causing that problem (unrelated to this PR), but we can deal with that later.  It will probably go away with the promise conversion anyway.